### PR TITLE
RAnalValue no longer refs an RRegItem ##analysis

### PR DESCRIPTION
**Pull request by trufae**
* Faster analysis and fix some leaks and uafs

- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/arch/p/m68k_cs/plugin.c [hunk 1]
```diff
@@ -204,23 +204,19 @@ static void opex(RStrBuf *buf, csh handle, cs_insn *insn) {
...
+static const char *parse_reg_name(csh handle, cs_insn *insn, int reg_num) {
 	switch (OPERAND (reg_num).type) {
 	case M68K_OP_REG:
-		reg->name = (char *)cs_reg_name (handle, OPERAND (reg_num).reg);
-		break;
+		return (char *)cs_reg_name (handle, OPERAND (reg_num).reg);
 	case M68K_OP_MEM:
 		if (OPERAND (reg_num).mem.base_reg != M68K_REG_INVALID) {
-			reg->name = (char *)cs_reg_name (handle, OPERAND (reg_num).mem.base_reg);
+			return (char *)cs_reg_name (handle, OPERAND (reg_num).mem.base_reg);
```
Comment (luc-tielen) [inline libr/arch/p/m68k_cs/plugin.c:213]:
```suggestion
			return cs_reg_name (handle, OPERAND (reg_num).mem.base_reg);
```

File: libr/arch/p/ppc_cs/plugin.c [hunk 2]
```diff
@@ -433,41 +433,31 @@ static int analop_vle(RArchSession *as, RAnalOp *op, ut64 addr, const ut8 *buf,
...
-			reg->name = (char *)cs_reg_name (handle, INSOP (reg_num).mem.base);
+			return cs_reg_name (handle, INSOP (reg_num).mem.base);
 		}
 		break;
-	default :
+	default:
 		break;
 	}
-	return 0;
+	return NULL;
```
Comment (luc-tielen) [inline libr/arch/p/ppc_cs/plugin.c:448]: move return to default case

File: libr/include/r_arch.h [hunk 3]
```diff
@@ -146,13 +136,11 @@ typedef struct r_arch_t {
 } RArch;
 
 typedef struct r_arch_session_t {
-#if R2_590
 	char *name; // used by .use to chk if it was set already
 	// TODO: name it "peer" instead of encoder. so the encoder can back reference the decoder
```
Comment (luc-tielen) [inline libr/include/r_arch.h:140]:
```suggestion
```

File: libr/include/r_arch.h [hunk 4]
```diff
@@ -271,6 +252,13 @@ R_API void r_arch_op_fini(RAnalOp *op);
 R_API void r_arch_op_free(void *_op);
 #endif
 
+#if 1
+// Deprecate!
```
Comment (luc-tielen) [inline libr/include/r_arch.h:256]:
```suggestion
// R2_590 Deprecate!
```
(or some other kind of TODO so we don't forget)

**Review by luc-tielen — CHANGES_REQUESTED** [review]

File: libr/anal/fcn.c [hunk 5]
```diff
@@ -233,7 +234,8 @@ static bool is_delta_pointer_table(ReadAhead *ra, RAnal *anal, RAnalFunction *fc
 	    && mov_aop.disp && mov_aop.disp != UT64_MAX) {
 		// disp in this case should be tbl_loc_off
 		*jmptbl_addr += mov_aop.disp;
-		if (o_reg_dst && reg_src && o_reg_dst->offset == reg_src->offset && omov_aop.disp != UT64_MAX) {
+		// if (o_reg_dst && reg_src && o_reg_dst->offset == reg_src->offset && omov_aop.disp != UT64_MAX) {
+		if (o_reg_dst && reg_src && !strcmp (o_reg_dst, reg_src) && omov_aop.disp != UT64_MAX) {
```
[the remark this answers is not in the document]
Comment (trufae) [reply libr/anal/fcn.c:238]: maybe worth doing r_str_eq as an inline function

File: libr/include/r_arch.h (hunk 3 above)
Comment (trufae) [reply libr/include/r_arch.h:140]: if i remove this comment then i need to change the variable below and more stuff. better keep that todo for another pr

File: libr/anal/p/anal_arm_cs.c [hunk 6]
```diff
@@ -4267,31 +4267,27 @@ static bool is_valid(arm_reg reg) {
 }
 
 // XXX this function is a disaster
```
[the remark this answers is not in the document]
Comment (trufae) [reply libr/anal/p/anal_arm_cs.c:4269]: its still a disaster :D but cant be fixed in this pr

**Pull request merged**